cocoa/src/toga_cocoa/widgets/webview.py

```diff
@@ -19,6 +19,40 @@ def _completion_handler(res: objc_id, error: objc_id) -> None:
     return _completion_handler
 
 
+def cookies_completion_handler(result):
+    def _completion_handler(cookies: objc_id) -> None:
+        try:
+            # Convert cookies from Objective-C to Python objects
+            cookies_array = py_from_ns(cookies)
+
+            # Structure the cookies as a list of dictionaries
+            structured_cookies = []
+            for cookie in cookies_array:
+                structured_cookies.append(
+                    {
```

Comment:
This suggests to me that there should be a `Cookie` object - preferably a `http.cookie.SimpleCookie` (or similar) to maximise compatibility with the rest of the Python ecosystem. It might also be possible to use a `http.cookiejar.CookieJar` for `structured_cookies`, but I'm less convinced that will be of value, as a CookieJar involves a lot of other cookie management that we likely don't need here.

Reply:
Personally I lean more into using http.cookiejar.CookieJar since I need to be able to access cookies with the same name but different domain. I'll see if I can get this implementend.